beetsplug/play.py

```diff
@@ -117,35 +111,76 @@ def play_music(self, lib, opts, args):
+        cancel = self._print_info(selection, command_str, open_args, item_type)
+
+        # Otherwise proceed with play command.
+        if not cancel:
+            play(command_str, paths, open_args)
+
+    def _create_command_str(self, args=None):
+        """Creates a command string from the config command and optional args.
+        """
+        command_str = config['play']['command'].get()
+        if not command_str:
+            return util.open_anything()
+        # Add optional arguments to the player command.
+        if args:
+            if ARGS_MARKER in command_str:
+                return command_str.replace(ARGS_MARKER, args)
+            else:
+                return u"{} {}".format(command_str, args)
+        else:
+            # Don't include the marker in the command.
+            return command_str.replace(" " + ARGS_MARKER, "")
+
+    def _playlist_or_paths(self, paths):
+        """Returns either the raw paths of items or a playlist of the items.
+        """
+        raw = config['play']['raw'].get(bool)
+        if raw:
+            return paths
+        else:
+            return [self._create_tmp_playlist(paths)]
+
+    def _print_info(self, selection, command_str, open_args,
+                    item_type='track'):
+        """Prompts user whether to continue if playlist exceeds threshold.
+        """
+        warning_threshold = config['play']['warning_threshold'].get(int)
+        # We use -2 as a default value for warning_threshold to detect if it is
+        # set or not. We can't use a falsey value because it would have an
+        # actual meaning in the configuration of this plugin, and we do not use
+        # -1 because some people might use it as a value to obtain no warning,
+        # which wouldn't be that bad of a practice.
+        if warning_threshold == -2:
+            # if warning_threshold has not been set by user, look for
+            # warning_treshold, to preserve backwards compatibility. See #1803.
+            # warning_treshold has the correct default value of 100.
+            warning_threshold = config['play']['warning_treshold'].get(int)
+
         # Warn user before playing any huge playlists.
+        item_type += 's' if len(selection) > 1 else ''
         if warning_threshold and len(selection) > warning_threshold:
             ui.print_(ui.colorize(
                 'text_warning',
                 u'You are about to queue {0} {1}.'.format(
                     len(selection), item_type)))
 
             if ui.input_options((u'Continue', u'Abort')) == 'a':
-                return
+                return True
 
+        # Print number of tracks or albums to be played, log command to be run.
         ui.print_(u'Playing {0} {1}.'.format(len(selection), item_type))
-        if raw:
-            open_args = paths
-        else:
-            open_args = [self._create_tmp_playlist(paths)]
-
         self._log.debug(u'executing command: {} {!r}', command_str, open_args)
```

Comment:
Maybe these logging messages should go in the method that actually plays the music?